Corzbank.Services/CardService.cs

```diff
@@ -36,6 +46,12 @@ public async Task<Card> GetCardById(Guid id)
 
         public async Task<Card> CreateCard(CardModel card)
         {
+            var currentUserEmail = _httpContextAccessor.HttpContext.User.Identity.Name;
+            var currentUser = await _userManager.FindByEmailAsync(currentUserEmail);
+
+            if (await _genericService.FindByCondition(c => c.CardType.Equals(card.CardType) && c.User.Id == currentUser.Id && c.IsActive) != null)
```

Comment:
move to a variable, complicated if-statement

Reply:
resolved